### DISCUSSION TITLE
用bot进行压测时，配置不强制内网，配置的地址也是外网（外网通过反向代理到bot的本机），bot一直显示超时，无法登录

### DISCUSSION THREAD

**Discussion by weiwsy** [kbengine/kbengine#1603] [General]
用bot进行压测时，配置不强制内网，配置的地址也是外网（外网通过反向代理到bot的本机），bot一直显示超时，无法登录
错误信息：bots01: [ERROR]: ClientObject::initLogin: recvfrom timeout!
配置如下

```
 <bots>
		<forceInternalLogin> false </forceInternalLogin>
		<host> 实际的外网域名 </host>
</bots>
```

改为`<forceInternalLogin> true </forceInternalLogin>` 后，可以登录了，但是iftop监测流量，发现如果只有机器人，没有玩家时，20015端口没有任何流量，启用一个玩家后，有了流量（流量基本等于每一个bot 0.7kb）。

**Comment by qhf566** [comment]
想找一款成熟的手游产品源码过来我们深度二次开发(全新换套美术，代码也都从新货混淆及优化整一遍)。意向最好是重度的mmo、arpg、角色扮演或动作手游，其他类型的游戏也可以，必须是高品质大厂近5年内做的产品，国内厂商或国外厂商开发的都可以，产品不能在国内运营过的。可以在国外运营过或运行中。准备了足够夸张的米给到你。如朋友那边有可以给推荐一下有米，不墨迹,鹅浩809996677。

**Comment by cnxkey** [comment]
您好！您所发的邮件已收到，谢谢！
